### PR TITLE
XnView

### PR DESCRIPTION
**Pull request by Tobias-B-Besemer**

### CODE REVIEW

File: pending/xnview.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+	CleanerML written by https://github.com/Tobias-B-Besemer
+	v0.1 done on 2018-12-19
+
+-->
+<cleaner id="xnview.new">
```
Comment (az0) [inline pending/xnview.xml]:
```suggestion
<cleaner id="xnview.new" os="windows">
```

File: pending/xnview.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  <label>XnView Classic for Windows</label>
+  <description>Picture Viewer</description>
+  <option id="mru">
+    <label>MRU Files and Folders</label>
+    <description>Delete list of MRU (Most recently used) Files and Folders</description>
+        <action command="ini" search="glob" path="$APPDATA\XnView\xnview.ini" name="FileName_0"/>
```
Comment (az0) [inline pending/xnview.xml]: Please change `search="glob"` to `file` for every `path` that does not use wildcards such as *

```suggestion
        <action command="ini" search="file" path="$APPDATA\XnView\xnview.ini" name="FileName_0"/>
```

File: pending/xnview.xml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        <action command="ini" search="glob" path="$APPDATA\XnView\xnview.ini" name="DirName_1"/>
+        <action command="ini" search="glob" path="$APPDATA\XnView\xnview.ini" name="DirName_2"/>
+        <action command="ini" search="glob" path="$APPDATA\XnView\xnview.ini" name="DirName_3"/>
+        <action command="ini" search="glob" path="$APPDATA\XnView\xnview.ini" name="DirName_4"/>
+        <action command="ini" search="glob" path="$APPDATA\XnView\xnview.ini" name="DirName_5"/>
+        <action command="ini" search="glob" path="$APPDATA\XnView\xnview.ini" name="DirName_6"/>
+        <action command="ini" search="glob" path="$APPDATA\XnView\xnview.ini" name="DirName_7"/>
+        <action command="ini" search="glob" path="$APPDATA\XnView\xnview.ini" name="DirName_8"/>
+        <action command="ini" search="glob" path="$APPDATA\XnView\xnview.ini" name="DirName_9"/>
+        <action command="ini" search="glob" path="%ProgramFiles(x86)%\XnView\xnview.ini" name="FileName_*"/>
```
Comment (az0) [inline pending/xnview.xml]: BleachBit 2.0 does not support wildcards in `name=`

**Comment by Tobias-B-Besemer** [comment]
Solved!

File: pending/xnview.xml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    <action command="ini" search="file" path="%WinDir%\xnview.ini" name="DirName_2"/>
+    <action command="ini" search="file" path="%WinDir%\xnview.ini" name="DirName_3"/>
+    <action command="ini" search="file" path="%WinDir%\xnview.ini" name="DirName_4"/>
+    <action command="ini" search="file" path="%WinDir%\xnview.ini" name="DirName_5"/>
+    <action command="ini" search="file" path="%WinDir%\xnview.ini" name="DirName_6"/>
+    <action command="ini" search="file" path="%WinDir%\xnview.ini" name="DirName_7"/>
+    <action command="ini" search="file" path="%WinDir%\xnview.ini" name="DirName_8"/>
+    <action command="ini" search="file" path="%WinDir%\xnview.ini" name="DirName_9"/>
+    <action command="winreg" path="HKCU\Software\XnView\Start\FileName_*"/>
+    <action command="winreg" path="HKCU\Software\XnView\Start\DirName_*"/>
```
Comment (az0) [inline pending/xnview.xml]: Sorry, `command="winreg"` does not support wildcards

File: pending/xnview.xml (hunk 1 above)
Comment (Tobias-B-Besemer) [reply pending/xnview.xml]: Fixed!

File: pending/xnview.xml [hunk 5]
```diff
@@ -61,12 +61,14 @@
         <action command="delete" search="file" path="$APPDATA\XnView\XnView.db"/>
         <action command="delete" search="file" path="%ProgramFiles(x86)%\XnView\XnView.db"/>
         <action command="delete" search="glob" path="%ProgramFiles(x86)%\XnView\cache\*.db"/>
+        <action command="delete" search="file" path="%ProgramFiles%\XnView\cache\*.db"/>
```
[the remark this answers is not in the document]
Comment (Tobias-B-Besemer) [reply pending/xnview.xml]: Fixed!

File: pending/xnview.xml (hunk 4 above)
Comment (Tobias-B-Besemer) [reply pending/xnview.xml]: Fixed!

**Comment by az0** [comment]
I would refactor like this: https://github.com/bleachbit/bleachbit/commit/49ea56c736632b920d8694c2c9976cdaf1685831

**Pull request merged**